### ISSUE TITLE
casher:143:in `read': No such file or directory - .casher/paths (Errno::ENOENT) on build cache

### ISSUE THREAD

**Issue by kenorb** [travis-ci/travis-ci#6033]
I've the problem with CI build where PR one fails, but for PUSH it's working fine.

<img width="685" alt="screen shot 2016-05-10 at 23 29 11" src="https://cloud.githubusercontent.com/assets/266306/15164494/12aca6f8-1707-11e6-8fe2-efa1ce83e264.png">

The error for PR for each instance is (build [#128827215](https://travis-ci.org/EA31337/FX-MT-VM/builds/128827215)):

```
store build cache
/home/travis/.casher/bin/casher:143:in `read': No such file or directory - /home/travis/.casher/paths (Errno::ENOENT)
	from /home/travis/.casher/bin/casher:143:in `changed?'
	from /home/travis/.casher/bin/casher:116:in `push'
	from /home/travis/.casher/bin/casher:52:in `block in run'
	from /home/travis/.rvm/rubies/ruby-1.9.3-p551/lib/ruby/1.9.1/timeout.rb:69:in `timeout'
	from /home/travis/.casher/bin/casher:52:in `run'
	from /home/travis/.casher/bin/casher:233:in `<main>'
```

Same configuration for the same commit works for push (build: [#128827196](https://travis-ci.org/EA31337/FX-MT-VM/builds/128827196)).

This started happening recently.

My [`.travis.yml`](https://github.com/EA31337/FX-MT-VM/blob/77de9572795567540900c25752da0319a5213feb/.travis.yml) has cache as follows:

Related: #6022 (but this is different error), #3239 (as per same error [here](https://github.com/travis-ci/travis-ci/issues/3239#issuecomment-185302602) and [here](https://github.com/travis-ci/travis-ci/issues/3239#issuecomment-191188452)).

**Comment by BanzaiMan** [comment]
Thanks for reporting the issue. Ultimately, though, I believe this is the same issue as #6022. So I'm closing this as a duplicate.

I explain below what is happening here. This is something of an implementation detail, so feel free to skip it.

Both of these builds (https://travis-ci.org/EA31337/FX-MT-VM/jobs/128827216 and https://travis-ci.org/EA31337/FX-MT-VM/jobs/128827197) have the exact same problem (that `/home/travis/.casher/paths` does not exist—because there is a space *and* parentheses in the name of a directory to be cached), but the bug manifests differently.

The difference between the two builds is not one of Push vs PR, but one of "whether or not a cache is found". The one that passes does *not* find a cache, so we short circuit the logic to test whether or not there is a change in the cache; there is a change, so we proceed directly to create an archive, where [`tar` fails](https://travis-ci.org/EA31337/FX-MT-VM/jobs/128827197#L2940) the same way as #6022.

The failing one, on the other hand, finds a usable cache, and therefore needs to check whether or not there are changes. But the file to check against does not exist. Thus it shows the error message reported in this issue. Because [the `errexit` option is set](https://github.com/EA31337/FX-MT-VM/blob/77de9572795567540900c25752da0319a5213feb/.travis.yml#L40), the build halts immediately when the caching utility exits with non-zero error. If you didn't set `-e`, the build will proceed to fail the same way as #6022.

#3239 is completely unrelated.

**Comment by kenorb** [comment]
Thanks, so I believe this going to be fixed as part of #6005?
